AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.kt

```diff
@@ -339,60 +343,77 @@ class ModelFieldEditor :
         }
     }
 
-    /*
-     * Allows the user to select a number less than the number of fields in the current model to
-     * reposition the current field to
-     * Processing time is scales with number of items
+    /**
+     * Displays a repositioning dialog allowing the user to move a field to a new position.
      */
-    private fun repositionFieldDialog() {
-        fieldNameInput = FixedEditText(this).apply { focusWithKeyboard() }
-        fieldNameInput?.let { fieldNameInput ->
-            fieldNameInput.setRawInputType(InputType.TYPE_CLASS_NUMBER)
-            AlertDialog.Builder(this).show {
-                customView(view = fieldNameInput, paddingStart = 64, paddingEnd = 64, paddingTop = 32)
-                title(text = String.format(resources.getString(R.string.model_field_editor_reposition), 1, fieldsLabels.size))
-                positiveButton(R.string.dialog_ok) {
-                    val newPosition = fieldNameInput.text.toString()
-                    val pos: Int =
-                        try {
-                            newPosition.toInt()
-                        } catch (n: NumberFormatException) {
-                            Timber.w(n)
-                            fieldNameInput.error = resources.getString(R.string.toast_out_of_range)
-                            return@positiveButton
-                        }
-                    if (pos < 1 || pos > fieldsLabels.size) {
-                        fieldNameInput.error = resources.getString(R.string.toast_out_of_range)
-                    } else {
-                        // Input is valid, now attempt to modify
-                        try {
-                            getColUnsafe.modSchema()
-                            repositionField(pos - 1)
-                        } catch (e: ConfirmModSchemaException) {
-                            e.log()
+    private fun repositionField() {
+        repositionFieldDialog(this, fieldsLabels.size, currentPos + 1) { newPosition ->
+            if (newPosition == currentPos + 1) return@repositionFieldDialog
 
-                            // Handle mod schema confirmation
-                            val c = ConfirmationDialog()
-                            c.setArgs(resources.getString(R.string.full_sync_confirmation))
-                            val confirm =
-                                Runnable {
-                                    try {
-                                        getColUnsafe.modSchemaNoCheck()
-                                        repositionField(pos - 1)
-                                    } catch (e1: JSONException) {
-                                        throw RuntimeException(e1)
-                                    }
-                                }
-                            c.setConfirm(confirm)
-                            this@ModelFieldEditor.showDialogFragment(c)
+            Timber.w("moving card template %d to %d", currentPos, newPosition)
```

Comment:
Shouldn't be a warning:

```suggestion
            Timber.i("Repositioning field from %d to %d", currentPos, newPosition)
```